frontend/src/components/compiler/PresetSelect.tsx

```diff
@@ -21,23 +21,31 @@ export default function PresetSelect({ className, platform, presetId, setPreset,
     setPreset: (preset: api.Preset) => void
     serverPresets?: api.Preset[]
 }) {
-    if (!serverPresets)
+    if (typeof serverPresets === "undefined")
         serverPresets = api.usePresets(platform)
 
-    const selectedPreset = serverPresets.find(p => p.id === presetId)
+    const sortedPresets = typeof serverPresets === "undefined" ? null : serverPresets.slice().sort((a, b) => a.name.localeCompare(b.name))
 
-    if (serverPresets.length > 0 && typeof presetId === "number" && !selectedPreset)
+    if (sortedPresets === null) {
+        return <Select
+            className={className}
+            options={{ "Loading": "Loading..." }}
+            value={"Loading"}
+            onChange={null}
+        />
+    }
+
+    const selectedPreset = sortedPresets.find((p: api.Preset) => p.id === presetId)
+
+    if (sortedPresets.length > 0 && typeof presetId === "number" && !selectedPreset)
         console.warn(`Scratch.preset == '${presetId}' but no preset with that id was found.`)
 
     return <Select
         className={className}
-        options={presetsToOptions(serverPresets)}
+        options={presetsToOptions(sortedPresets)}
         value={selectedPreset?.name || "Custom"}
         onChange={name => {
-            const p = serverPresets.find(p => p.name === name)
-            if (p) {
-                setPreset(p)
-            }
+            setPreset(name === "Custom" ? null : sortedPresets.find((p: api.Preset) => p.name === name))
```

Comment:
Passing null into setPreset was the original issue (the setPreset method was trying to access fields of null and erroring)

Reply:
Can you demonstrate this on the vercel deployment? I'll take another look but I tested changing the preset to Custom & saving and everything behaved as expected

Reply:
If I go to the new scratch page and toggle between something and custom, I get

<img width="314" alt="image" src="https://github.com/decompme/decomp.me/assets/2985314/2e1632b4-3edd-40ab-8582-fa6d6c823bd3">

Reply:
Aha. I've been testing the "Custom" on the Scratch Options tab which is where it works. Let me figure out a fix for the New Scratch page 👍